rust/crd/src/lib.rs

```diff
@@ -184,33 +184,25 @@ pub struct HdfsClusterConfig {
     pub rack_awareness: Option<Vec<TopologyLabel>>,
 }
 
-#[derive(Clone, Debug, Default, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
+#[derive(Clone, Debug, Deserialize, Eq, Hash, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct TopologyLabel {
-    /// Name of the label type that will be typically either `node` or `pod`, used to create a
-    /// topology out of datanodes.
-    label_type: TopologyLabelType,
-
-    /// Name of the label that will be used to resolve a datanode to a topology zone.
-    label_name: String,
+pub enum TopologyLabel {
+    /// Name of the label on the Kubernetes Node (where the Pod is placed on) used to resolve a datanode to a topology
+    /// zone.
+    NodeLabel(String),
+    /// Name of the label on the Kubernetes Pod used to resolve a datanode to a topology zone.
+    PodLabel(String),
```

Comment:
code-style: https://docs.stackable.tech/home/nightly/contributor/code-style-guide#_formatting_of_struct_fields_and_enum_variants

Fields or Variants with comments/attributes need empty lines between.

```suggestion
pub enum TopologyLabel {
    /// Name of the label on the Kubernetes Node (where the Pod is placed on) used to resolve a datanode to a topology
    /// zone.
    NodeLabel(String),
    
    /// Name of the label on the Kubernetes Pod used to resolve a datanode to a topology zone.
    PodLabel(String),
```

Reply:
Sure, forgot that :see_no_evil: